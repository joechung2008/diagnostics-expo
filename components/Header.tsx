import { useApp } from '@/components/AppContext';
import { useTheme } from '@/components/ThemeContext';
import { Environment, getEnvironmentName } from '@/lib/environment';
import { Ionicons } from '@expo/vector-icons';
import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';

export default function Header() {
  const { colors } = useTheme();
  const {
    environment,
    showEnvironmentDropdown,
    setShowEnvironmentDropdown,
    handleEnvironmentSelect,
    showPaasServerless,
    handlePaasServerlessPress,
    handleWebsitesPress,
  } = useApp();

  const dynamicStyles = {
    titleBar: {
      alignItems: 'center' as const,
      justifyContent: 'center' as const,
      minHeight: 44,
      padding: 10,
      backgroundColor: colors.surface,
      borderBottomWidth: 1,
      borderBottomColor: colors.border,
    },
    toolbarTitle: {
      fontSize: 18,
      fontWeight: '600' as const,
      color: colors.text,
      textAlign: 'center' as const,
    },
    controlsBar: {
      alignItems: 'center' as const,
      flexDirection: 'row' as const,
      justifyContent: 'space-between' as const,
      minHeight: 44,
      padding: 10,
      backgroundColor: colors.surface,
      borderBottomWidth: 1,
      borderBottomColor: colors.border,
    },
    environmentButton: {
      alignSelf: 'center' as const,
      padding: 10,
      minWidth: 120,
      backgroundColor: colors.surfaceSecondary,
      borderRadius: 4,
      flexDirection: 'row' as const,
      justifyContent: 'space-between' as const,
      alignItems: 'center' as const,
    },
    toolbarRight: {
      alignItems: 'center' as const,
      flexDirection: 'row' as const,
    },
    environmentButtonText: {
      fontSize: 14,
      color: colors.text,
    },
    environmentDropdown: {
      position: 'absolute' as const,
      top: 50,
      left: 10,
      right: 10,
      backgroundColor: colors.surface,
      borderWidth: 1,
      borderColor: colors.border,
      borderRadius: 4,
      zIndex: 1000,
      maxHeight: 200,
      minHeight: 50,
      shadowColor: colors.text,
      shadowOffset: { width: 0, height: 2 },
      shadowOpacity: 0.1,
      shadowRadius: 4,
      elevation: 5,
    },
    environmentOption: {
      padding: 14,
      backgroundColor: colors.surface,
      borderBottomWidth: 1,
      borderBottomColor: colors.borderSecondary,
    },
    environmentOptionLast: {
      borderBottomWidth: 0,
    },
    environmentOptionText: {
      fontSize: 16,
      color: colors.text,
    },
    toolbarButton: {
      padding: 16,
    },
    toolbarButtonText: {
      fontSize: 14,
      color: colors.primary,
    },
  };

  return (
    <>
      <View style={dynamicStyles.titleBar}>
        <Text style={dynamicStyles.toolbarTitle}>
          Azure Portal Extensions Dashboard
        </Text>
      </View>
      <View style={dynamicStyles.controlsBar}>
        <TouchableOpacity
          style={dynamicStyles.environmentButton}
          onPress={() => setShowEnvironmentDropdown(!showEnvironmentDropdown)}
        >
          <Text style={dynamicStyles.environmentButtonText}>
            {getEnvironmentName(environment)}
          </Text>
          <Ionicons name="chevron-down" size={16} color={colors.text} />
        </TouchableOpacity>
        <View style={dynamicStyles.toolbarRight}>
          {showPaasServerless && (
            <TouchableOpacity
              style={dynamicStyles.toolbarButton}
              onPress={handlePaasServerlessPress}
            >
              <Text style={dynamicStyles.toolbarButtonText}>
                paasserverless
              </Text>
            </TouchableOpacity>
          )}
          <TouchableOpacity
            style={dynamicStyles.toolbarButton}
            onPress={handleWebsitesPress}
          >
            <Text style={dynamicStyles.toolbarButtonText}>websites</Text>
          </TouchableOpacity>
        </View>
      </View>

      {showEnvironmentDropdown && (
        <View style={dynamicStyles.environmentDropdown}>
          {Object.values(Environment).map((env, index) => (
            <TouchableOpacity
              key={env}
              style={[
                dynamicStyles.environmentOption,
                index === Object.values(Environment).length - 1 &&
                  dynamicStyles.environmentOptionLast,
              ]}
              onPress={() => handleEnvironmentSelect(env)}
            >
              <Text style={dynamicStyles.environmentOptionText}>
                {getEnvironmentName(env)}
              </Text>
            </TouchableOpacity>
          ))}
        </View>
      )}
    </>
  );
}
