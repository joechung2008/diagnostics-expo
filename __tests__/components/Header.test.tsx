import Header from '@/components/Header';
import { fireEvent, render } from '@testing-library/react-native';

// Mock the ThemeContext
jest.mock('@/components/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      surface: '#f5f5f5',
      surfaceSecondary: '#e0e0e0',
      text: '#000000',
      border: '#cccccc',
    },
  }),
}));

// Create mocks for useApp functions
const mockSetShowEnvironmentDropdown = jest.fn();
const mockHandleEnvironmentSelect = jest.fn();
const mockHandlePaasServerlessPress = jest.fn();
const mockHandleWebsitesPress = jest.fn();
const mockUseApp = jest.fn();

// Mock the AppContext
jest.mock('@/components/AppContext', () => ({
  useApp: () => mockUseApp(),
}));

// Mock expo/vector-icons
jest.mock('@expo/vector-icons', () => ({
  Ionicons: 'MockIonicons',
}));

describe('Header', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    mockUseApp.mockReturnValue({
      environment: 'https://hosting.portal.azure.net/api/diagnostics',
      setShowEnvironmentDropdown: mockSetShowEnvironmentDropdown,
      showEnvironmentDropdown: false,
      showPaasServerless: true,
      handleEnvironmentSelect: mockHandleEnvironmentSelect,
      handlePaasServerlessPress: mockHandlePaasServerlessPress,
      handleWebsitesPress: mockHandleWebsitesPress,
    });
  });

  it('renders with default state', () => {
    const { toJSON } = render(<Header />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with environment dropdown visible', () => {
    mockUseApp.mockReturnValueOnce({
      environment:
        'https://hosting.azureportal.usgovcloudapi.net/api/diagnostics',
      setShowEnvironmentDropdown: mockSetShowEnvironmentDropdown,
      showEnvironmentDropdown: true,
      showPaasServerless: false,
      handleEnvironmentSelect: mockHandleEnvironmentSelect,
      handlePaasServerlessPress: mockHandlePaasServerlessPress,
      handleWebsitesPress: mockHandleWebsitesPress,
    });

    const { toJSON } = render(<Header />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with different environment', () => {
    mockUseApp.mockReturnValueOnce({
      environment:
        'https://hosting.azureportal.chinacloudapi.cn/api/diagnostics',
      setShowEnvironmentDropdown: mockSetShowEnvironmentDropdown,
      showEnvironmentDropdown: false,
      showPaasServerless: true,
      handleEnvironmentSelect: mockHandleEnvironmentSelect,
      handlePaasServerlessPress: mockHandlePaasServerlessPress,
      handleWebsitesPress: mockHandleWebsitesPress,
    });

    const { toJSON } = render(<Header />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with PaaS Serverless hidden', () => {
    mockUseApp.mockReturnValueOnce({
      environment: 'https://hosting.portal.azure.net/api/diagnostics',
      setShowEnvironmentDropdown: mockSetShowEnvironmentDropdown,
      showEnvironmentDropdown: false,
      showPaasServerless: false,
      handleEnvironmentSelect: mockHandleEnvironmentSelect,
      handlePaasServerlessPress: mockHandlePaasServerlessPress,
      handleWebsitesPress: mockHandleWebsitesPress,
    });

    const { toJSON } = render(<Header />);
    expect(toJSON()).toMatchSnapshot();
  });

  describe('interactions', () => {
    it('calls setShowEnvironmentDropdown when environment button is pressed', () => {
      const { getByText } = render(<Header />);

      const environmentButton = getByText('Public Cloud');
      fireEvent.press(environmentButton);

      expect(mockSetShowEnvironmentDropdown).toHaveBeenCalledWith(true);
    });

    it('toggles environment dropdown when button is pressed multiple times', () => {
      mockUseApp.mockReturnValue({
        environment: 'https://hosting.portal.azure.net/api/diagnostics',
        setShowEnvironmentDropdown: mockSetShowEnvironmentDropdown,
        showEnvironmentDropdown: true,
        showPaasServerless: true,
        handleEnvironmentSelect: mockHandleEnvironmentSelect,
        handlePaasServerlessPress: mockHandlePaasServerlessPress,
        handleWebsitesPress: mockHandleWebsitesPress,
      });

      const { getAllByText } = render(<Header />);

      const environmentButtons = getAllByText('Public Cloud');
      fireEvent.press(environmentButtons[0]);

      expect(mockSetShowEnvironmentDropdown).toHaveBeenCalledWith(false);
    });

    it('calls handleEnvironmentSelect when environment option is pressed', () => {
      mockUseApp.mockReturnValue({
        environment: 'https://hosting.portal.azure.net/api/diagnostics',
        setShowEnvironmentDropdown: mockSetShowEnvironmentDropdown,
        showEnvironmentDropdown: true,
        showPaasServerless: true,
        handleEnvironmentSelect: mockHandleEnvironmentSelect,
        handlePaasServerlessPress: mockHandlePaasServerlessPress,
        handleWebsitesPress: mockHandleWebsitesPress,
      });

      const { getByText } = render(<Header />);

      const fairfaxOption = getByText('Fairfax');
      fireEvent.press(fairfaxOption);

      expect(mockHandleEnvironmentSelect).toHaveBeenCalledWith(
        'https://hosting.azureportal.usgovcloudapi.net/api/diagnostics'
      );
    });

    it('calls handlePaasServerlessPress when paasserverless button is pressed', () => {
      const { getByText } = render(<Header />);

      const paasButton = getByText('paasserverless');
      fireEvent.press(paasButton);

      expect(mockHandlePaasServerlessPress).toHaveBeenCalled();
    });

    it('calls handleWebsitesPress when websites button is pressed', () => {
      const { getByText } = render(<Header />);

      const websitesButton = getByText('websites');
      fireEvent.press(websitesButton);

      expect(mockHandleWebsitesPress).toHaveBeenCalled();
    });
  });
});
