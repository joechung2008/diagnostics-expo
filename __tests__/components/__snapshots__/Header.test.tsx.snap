// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders with PaaS Serverless hidden 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f5f5f5",
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 1,
        "justifyContent": "center",
        "minHeight": 44,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 18,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Azure Portal Extensions Dashboard
    </Text>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f5f5f5",
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minHeight": 44,
        "padding": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#e0e0e0",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "minWidth": 120,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 14,
          }
        }
      />
      <MockIonicons
        color="#000000"
        name="chevron-down"
        size={16}
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": undefined,
              "fontSize": 14,
            }
          }
        >
          websites
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`Header renders with default state 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f5f5f5",
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 1,
        "justifyContent": "center",
        "minHeight": 44,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 18,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Azure Portal Extensions Dashboard
    </Text>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f5f5f5",
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minHeight": 44,
        "padding": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#e0e0e0",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "minWidth": 120,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 14,
          }
        }
      />
      <MockIonicons
        color="#000000"
        name="chevron-down"
        size={16}
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": undefined,
              "fontSize": 14,
            }
          }
        >
          paasserverless
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": undefined,
              "fontSize": 14,
            }
          }
        >
          websites
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`Header renders with different environment 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f5f5f5",
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 1,
        "justifyContent": "center",
        "minHeight": 44,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 18,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Azure Portal Extensions Dashboard
    </Text>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f5f5f5",
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minHeight": 44,
        "padding": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#e0e0e0",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "minWidth": 120,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 14,
          }
        }
      />
      <MockIonicons
        color="#000000"
        name="chevron-down"
        size={16}
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": undefined,
              "fontSize": 14,
            }
          }
        >
          paasserverless
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": undefined,
              "fontSize": 14,
            }
          }
        >
          websites
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`Header renders with environment dropdown visible 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f5f5f5",
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 1,
        "justifyContent": "center",
        "minHeight": 44,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 18,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Azure Portal Extensions Dashboard
    </Text>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f5f5f5",
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minHeight": 44,
        "padding": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#e0e0e0",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "minWidth": 120,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 14,
          }
        }
      />
      <MockIonicons
        color="#000000"
        name="chevron-down"
        size={16}
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": undefined,
              "fontSize": 14,
            }
          }
        >
          websites
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderColor": "#cccccc",
        "borderRadius": 4,
        "borderWidth": 1,
        "elevation": 5,
        "left": 10,
        "maxHeight": 200,
        "minHeight": 50,
        "position": "absolute",
        "right": 10,
        "shadowColor": "#000000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 4,
        "top": 50,
        "zIndex": 1000,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#f5f5f5",
          "borderBottomColor": undefined,
          "borderBottomWidth": 1,
          "opacity": 1,
          "padding": 14,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
          }
        }
      >
        Public Cloud
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#f5f5f5",
          "borderBottomColor": undefined,
          "borderBottomWidth": 1,
          "opacity": 1,
          "padding": 14,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
          }
        }
      >
        Fairfax
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#f5f5f5",
          "borderBottomColor": undefined,
          "borderBottomWidth": 0,
          "opacity": 1,
          "padding": 14,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
          }
        }
      >
        Mooncake
      </Text>
    </View>
  </View>,
]
`;
